File: tests/cpp/signal_processing_demo/src/windowing.cpp
#include <iostream>
#include <stdlib.h>

#ifdef _WIN32
#include <windows.h>
#else
#include <unistd.h>
#endif

#include "board_shim.h"
#include "data_filter.h"

using namespace std;


int main (int argc, char *argv[])
{
    int res = 0;
    try
    {
        double *window_data = NULL;
        int window_len = 20;
        for (int window_function = 0; window_function < 4; window_function++)
        {
            std::cout << "Current window function ID: " << window_function << std::endl;
            std::cout << "Window data :" << std::endl;

            // just apply different windowing algorithms and print results
            switch (window_function)
            {
                case 0:
                    window_data = DataFilter::get_window (window_function, window_len);
                    break;
                case 1:
                    window_data = DataFilter::get_window (window_function, window_len);
                    break;
                case 2:
                    window_data = DataFilter::get_window (window_function, window_len);
                    break;
                case 3:
                    window_data = DataFilter::get_window (window_function, window_len);
                    break;
            }
            std::cout << window_data << std::endl;
            delete[] window_data;
        }
    }
    catch (const BrainFlowException &err)
    {
        BoardShim::log_message ((int)LogLevels::LEVEL_ERROR, err.what ());
        res = err.exit_code;
    }

    return res;
}

File: python-package/examples/tests/windowing.py
# import time

from brainflow.data_filter import DataFilter, WindowOperations


def main():
    # demo for data windowing
    window_len = 20

    for window_function in range(4):

        if window_function == WindowOperations.NO_WINDOW.value:
            print('Window data for NO_WINDOW function:')
            window_data = DataFilter.get_window(WindowOperations.NO_WINDOW.value, window_len)
        elif window_function == WindowOperations.HANNING.value:
            print('Window data for HANNING function:')
            window_data = DataFilter.get_window(WindowOperations.HANNING.value, window_len)
        elif window_function == WindowOperations.HAMMING.value:
            print('Window data for HAMMING function:')
            window_data = DataFilter.get_window(WindowOperations.HAMMING.value, window_len)
        else:
            print('Window data for BLACKMAN_HARRIS function:')
            window_data = DataFilter.get_window(WindowOperations.BLACKMAN_HARRIS.value, window_len)

        print(window_data)


if __name__ == "__main__":
    main()
